File: 3396_minimum_number_of_operations_to_make_elements_in_array_distinct.cpp
class Solution {
public:
    int minimumOperations(vector<int>& nums) {
        unordered_set<int> s(nums.begin(), nums.end());
        if (s.size() == nums.size()) return 0;  // Already distinct
        
        int operations = 0;
        
        while (nums.size() > 3) {
            nums.erase(nums.begin(), nums.begin() + 3);
            operations++;
            
            unordered_set<int> tempSet(nums.begin(), nums.end());
            if (tempSet.size() == nums.size()) break;  // Distinct elements found
        }
        
        unordered_set<int> finalSet(nums.begin(), nums.end());
        return operations + (finalSet.size() < nums.size());
    }
};

File: 3396_minimum_number_of_operations_to_make_elements_in_array_distinct.java
import java.util.HashSet;

class Solution {
    public int minimumOperations(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }
        if (set.size() == nums.length) {
            return 0;
        }
        int operations = 0;
        while (nums.length > 3) {
            int[] temp = new int[nums.length - 3];
            System.arraycopy(nums, 3, temp, 0, nums.length - 3);
            nums = temp;
            operations++;
            set.clear();
            for (int num : nums) {
                set.add(num);
            }
            if (set.size() == nums.length) {
                break;
            }
        }
        set.clear();
        for (int num : nums) {
            set.add(num);
        }
        return operations + (set.size() < nums.length ? 1 : 0);
    }
}
